src/plist.h

```diff
@@ -26,38 +26,86 @@
 #ifndef GAP_PLIST_H
 #define GAP_PLIST_H
 
+#include <src/debug.h> /* for GAP_ASSERT */
 
 /****************************************************************************
 **
-
 *F  NEW_PLIST(<type>,<plen>)  . . . . . . . . . . . allocate a new plain list
 **
 **  'NEW_PLIST'  allocates    a new plain   list  of  type <type> ('T_PLIST',
 **  'T_SET', 'T_VECTOR') that has room for at least <plen> elements.
 **
-**  Note that 'NEW_PLIST' is a  macro, so do not call  it with arguments that
-**  have side effects.
 */
-#define NEW_PLIST(type,plen)            NewBag(type,((plen)+1)*sizeof(Obj))
+static inline Obj NEW_PLIST(UInt type, Int plen)
+{
+    GAP_ASSERT(plen >= 0);
+    return NewBag(type, (plen + 1) * sizeof(Obj));
+}
 
+/****************************************************************************
+**
+*F  IS_PLIST( <list> )  . . . . . . . . . . . check if <list> is a plain list
+*/
+static inline Int IS_PLIST(Obj list)
+{
+    return FIRST_PLIST_TNUM <= TNUM_OBJ(list) &&
+           TNUM_OBJ(list) <= LAST_PLIST_TNUM;
+}
 
 /****************************************************************************
 **
-*F  GROW_PLIST(<list>,<plen>) . . . .  make sure a plain list is large enough
+*F  IS_PLIST_OR_POSOBJ( <list> )  . . . . . . .check type of <list>
 **
-**  'GROW_PLIST' grows  the plain list <list>  if necessary to ensure that it
-**  has room for at least <plen> elements.
+**  Checks if this is 'PLIST'-like.
+**  This function is used in a GAP_ASSERT checking if calling functions like
+**  SET_ELM_PLIST is acceptable on an Obj.
+**  We also check is the 'COPYING' bit is set (which is not checked by
```

Comment:
First "is" should be "if". Also, the whole sentence conveys little useful info... anybody looking at the code can see this; more interesting is the *why*, which is missing. How about this: "Unlike IS_PLIST, this function also accepts plists which are being copied (and hence have the COPYING flag set), as well as positional objects (which have the same memory layout as plists), as the plist APIs using it for assertion checks are in practice invoked on such objects, too."